### PR TITLE
Bug 1825124 - Change the way we show the post-install popup once an extension has been installed.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1825124

### CODE REVIEW

File: android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt [hunk 1]
```diff
@@ -229,6 +229,13 @@ object WebExtensionSupport {
 
                 override fun onInstalled(extension: WebExtension) {
                     registerInstalledExtension(store, extension)
+                    if (!extension.isBuiltIn()) {
```
Comment (willdurand) [inline android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt:232]: I think we should add a comment here to explain why we filter out built-in extensions.

File: android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/extension/WebExtensionPromptRequest.kt [hunk 2]
```diff
@@ -17,12 +17,20 @@ sealed class WebExtensionPromptRequest(
...
+     * @property onConfirm A callback indicating whether the permissions where granted or not.
      */
     data class Permissions(
         override val extension: WebExtension,
         val onConfirm: (Boolean) -> Unit,
     ) : WebExtensionPromptRequest(extension)
+
+    /**
+     * Value type that represents a request for showing post-installation prompt.
+     * Normally used to give users an opportunity to enable the [extension] in private browser.
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/extension/WebExtensionPromptRequest.kt:30]: Thanks!
Addressed!

File: android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt (hunk 1 above)
Comment (Amejia481) [reply android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt:234]: Added!

File: android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonInstallationDialogFragment.kt [hunk 3]
```diff
@@ -110,6 +116,10 @@ class AddonInstallationDialogFragment : AppCompatDialogFragment() {
         iconJob?.cancel()
     }
 
+    override fun onCancel(dialog: DialogInterface) {
+        super.onCancel(dialog)
+        onDismissed?.invoke()
+    }
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonInstallationDialogFragment.kt:124]: Addressed!

**Comment by Amejia481** [comment]
Thanks @willdurand double checking!

I think I addressed all the comments in including the bug mentioned above.

**Pull request merged**